src/main/java/flight/reservation/flight/Flight.java

```diff
@@ -30,21 +31,12 @@ private void checkValidity() throws IllegalArgumentException {
 
     private boolean isAircraftValid(Airport airport) {
         return Arrays.stream(airport.getAllowedAircrafts()).anyMatch(x -> {
-            String model;
-            if (this.aircraft instanceof PassengerPlane) {
-                model = ((PassengerPlane) this.aircraft).model;
-            } else if (this.aircraft instanceof Helicopter) {
-                model = ((Helicopter) this.aircraft).getModel();
-            } else if (this.aircraft instanceof PassengerDrone) {
-                model = "HypaHype";
-            } else {
-                throw new IllegalArgumentException(String.format("Aircraft is not recognized"));
-            }
+            String model = this.aircraft.getModel();
             return x.equals(model);
```

Comment:
```suggestion
            return allowedAircraft.equals(model);
```
Don't forget it here.